clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -998,7 +958,7 @@ public List<PartitionInfo> partitionsFor(String topic, Duration timeout) {
             wakeupTrigger.setActiveTask(topicMetadataEvent.future());
             try {
                 Map<String, List<PartitionInfo>> topicMetadata =
-                        applicationEventHandler.addAndGet(topicMetadataEvent, timer);
+                        applicationEventHandler.addAndGet(topicMetadataEvent);
```

Comment:
The timer is only used in the event.

Reply:
Removed use of `Timer` in favor of calculating the deadline from the `time` and `timeout` directly.